src/raftstore/store/store.rs

```diff
@@ -1251,33 +1251,48 @@ impl<T: Transport, C: PdClient> Store<T, C> {
         }
     }
 
-    fn propose_raft_command(&mut self, msg: RaftCmdRequest, cb: Callback) {
-        let mut resp = RaftCmdResponse::new();
-
+    fn pre_propose_raft_command(&mut self,
+                                msg: RaftCmdRequest)
+                                -> Either<RaftCmdRequest, RaftCmdResponse> {
```

Comment:
First of all, I am trying to avoid `clone()` on `RaftCmdRequest`. There is a `execute_status_command` inside, so it has to take a `RaftCmdRequest`. If all check is okay, no response will be returned, and the caller should continue processing the request. By returning the request, it saves a `clone()` call.